substrate/frame/revive/src/exec.rs

```diff
@@ -1028,7 +1045,7 @@ where
 		let frame = self.top_frame();
 		let entry_point = frame.entry_point;
 		let delegated_code_hash =
-			if frame.delegate_caller.is_some() { Some(*executable.code_hash()) } else { None };
+			if frame.delegate.is_some() { Some(*executable.code_hash()) } else { None };
```

Comment:
I can't mark the code where this is used later down. But it has to be changed. Now that we are taking a `deposit_limit` we need to make sure to enforce it by calling `enforce_subcall_limit`.  Otherwise the limit you added will not be enforced. We will also need a test for that.